ARExperiment/ARExperiment/ARViewModel.swift

```diff
@@ -4,12 +4,24 @@ import ARKit
 
 class ARViewModel {
 
-    func createSceneNodeForAsset(_ assetName: String, assetPath: String) -> SCNNode? {
-        guard let paperPlaneScene = SCNScene(named: assetPath) else {
+    func createSceneNodeForAsset(_ nodeName: String, assetFolder: String? = nil, fileName: String, assetExtension: String) -> SCNNode? {
+        let pathName = createAssetPath(assetFolder: assetFolder, fileName: fileName, fileExtension: assetExtension)
+        guard let scene = SCNScene(named: pathName) else {
             return nil
         }
-        let carNode = paperPlaneScene.rootNode.childNode(withName: assetName, recursively: true)
-        return carNode
+        let node = scene.rootNode.childNode(withName: nodeName, recursively: true)
+        return node
+    }
+
+    func getTransformForAnchor(location: CGPoint, sceneView: ARSCNView, resultType: ARHitTestResult.ResultType) -> simd_float4x4? {
+        guard let hitPoint = getHitResults(location: location,
+                                           sceneView: sceneView,
+                                           resultType: resultType) else {
+                                            print("failed to find hit point")
+                                            return nil
+        }
+        let rotate = simd_float4x4(SCNMatrix4MakeRotation(sceneView.session.currentFrame!.camera.eulerAngles.y, 0, 1, 0))
+        return simd_mul(hitPoint.worldTransform, rotate)
     }
 
     func getHitResults(location: CGPoint, sceneView: ARSCNView, resultType: ARHitTestResult.ResultType) -> ARHitTestResult? {
```

Comment:
just a naming thing here:
- swift/objc functions tend not to have "get" in the name
- the name suggests it returns multiple results (getHitResult**S**) but it only returns one! 

also we could make this more fluent, by using internal/external parameter names:
```swift
hitResults(at location: CGPoint, in sceneView: ARSCNView, withType type: ARHitTestResult.ResultType)
```